subsys/bluetooth/mesh/main.c

```diff
@@ -108,7 +105,7 @@ int bt_mesh_provision(const uint8_t net_key[16], uint16_t net_idx,
 						    net_key);
 		if (err) {
 			LOG_ERR("Failed to import cdb network key");
-			goto end;
+			goto error_exit;
 		}
 		bt_mesh_cdb_subnet_store(subnet);
```

Comment:
This needs to be aborted if keys importing failed (or network creation failed, or if anything else in bt_mesh_provision failed).

Reply:
This is not correct since subnetwork is created in advance.

Reply:
This is not correct to store anything if the whole operation is aborted regardless of whether it has been created before or not.